engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -0,0 +1,769 @@
+        } finally {
+            unmountStore();
+        }
+    }
+
+    private static String calculateChecksum(MessageDigest digest, File file) {
+        try {
+            FileInputStream fis = new FileInputStream(file);
+            byte[] byteArray = new byte[1024];
+            int bytesCount = 0;
+
+            while ((bytesCount = fis.read(byteArray)) != -1) {
+                digest.update(byteArray, 0, bytesCount);
+            }
+
+            fis.close();
+            byte[] bytes = digest.digest();
+
+            StringBuilder sb = new StringBuilder();
+            for (byte aByte : bytes) {
+                sb.append(Integer
+                        .toString((aByte & 0xff) + 0x100, 16)
+                        .substring(1));
+            }
+            return sb.toString();
+        } catch (IOException e) {
+            String errMsg = String.format("Failed to calculate Checksum of template file: %s due to: %s ", file.getName(), e);
+            LOGGER.error(errMsg);
+            throw new CloudRuntimeException(errMsg);
+        }
+    }
+
+    public static long isTemplateAlreadyRegistered(Connection conn, Pair<Hypervisor.HypervisorType, String> hypervisorAndTemplateName) {
+        long templateId = -1;
+        try {
+            PreparedStatement pstmt = conn.prepareStatement("select id from `cloud`.`vm_template` where name = ? and removed is null order by id desc limit 1");
+            // Get systemvm template id for corresponding hypervisor
+            pstmt.setString(1, hypervisorAndTemplateName.second());
+            try (ResultSet rs = pstmt.executeQuery()) {
+                if (rs.next()) {
+                    templateId = rs.getLong(1);
+                }
+            } catch (final SQLException e) {
+                String errMsg = String.format("updateSystemVmTemplates: Exception caught while getting ids of SystemVM templates: %s ", e.getMessage());
+                LOGGER.error(errMsg);
+                throw new CloudRuntimeException(errMsg);
+            }
+        } catch (SQLException e) {
+            String errorMessage = "Unable to upgrade the database";
+            LOGGER.error(errorMessage, e);
+            throw new CloudRuntimeException(errorMessage, e);
+        }
+        return templateId;
+    }
+
+    private static String fetchTemplatesPath() {
+            String filePath = relativeTemplatePath + metadataFileName;
+            LOGGER.debug("Looking for file [" + filePath + "] in the classpath.");
+            File metaFile = new File(filePath);
+            String templatePath = null;
+            if (metaFile.exists()) {
+                templatePath = relativeTemplatePath;
+            }
+            if (templatePath == null) {
+                filePath = AbsolutetemplatesPath + metadataFileName;
+                metaFile = new File(filePath);
+                templatePath = AbsolutetemplatesPath;
+                LOGGER.debug("Looking for file [" + filePath + "] in the classpath.");
```

Comment:
The message here and in line 341 are equal, we could extract this message to a variable with '%s' and use `String.format` in each log.